self-core-impl/src/main/java/com/selfxdsd/core/BaseLogin.java

```diff
@@ -0,0 +1,46 @@
+package com.selfxdsd.core;
+
+import com.selfxdsd.api.Login;
+import com.selfxdsd.api.User;
+import com.selfxdsd.api.storage.Storage;
+
+/**
+ * Base login implementation that aids concrete implementations of Login.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.1
+ */
+abstract class BaseLogin implements Login {
+
+    @Override
+    public User signUp(final Storage storage) {
+        return signUp(storage, userCredentials(storage));
+    }
```

Comment:
@criske Metoda asta o poti face ``default``, pe interfata ``Login`` (dar trebuie sa pui in ea direct logica din ``private ... signUp(...)``. 

Cam asa:

```java
public interface Login {
//...
default User signUp(final Storage storage) {
        final User credentials = this.userCredentials(storage);
        User authenticated = storage.users().user(
            user.username(), user.provider().name()
        );
        if(authenticated == null) {
            authenticated = storage.users().signUp(user);
        }
        return authenticated;
    }
//...
}
```
Si atunci cred ca nu mai e nevoie de ``BaseLogin``.

Reply:
Ok, sa vad daca merge si asa.

Reply:
Hm. daca fac asta o sa trebuiasca sa scot "user()" din BaseLogin si sa o bag in Login, si asta ar cam incalca principiul "single-responsibility" al interfetei. Acum interfata ar avea doua roluri: de signUp si de builder/factory.